### PR TITLE
Add PyCGTOOL

### PR DESCRIPTION
**Pull request by jag1g13**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/main/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by jag1g13** [comment]
I've listed myself as the maintainer, and am obviously happy to be the maintainer since I submitted the PR. But just in case, here's a comment to confirm this :)

**Comment by jag1g13** [comment]
@conda-forge-admin, please ping team

**Comment by jag1g13** [comment]
I beleive this is ready for review, but this is the first package I've submitted to conda-forge, so please do let me know if there's anything I've missed

File: recipes/pycgtool/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+requirements:
+  build:
+    - {{ compiler('c') }}
+  host:
+    - pip
+    - python
+    - poetry
+  run:
+    - astunparse ==1.6.2
+    - cython >=0.29.21,<0.30.0
```
Comment (ocefpaf) [inline recipes/pycgtool/meta.yaml:29]: Is cython needed at run time? Usually it is only a host/build dependency.

Comment (jag1g13) [reply recipes/pycgtool/meta.yaml:29]: Hmm, no, on second thought it actually shouldn't be required at all for conda. It's a requirement of the `pip` version in order to satisfy one of the other dependencies, but that dependency is already built in this case

Comment (ocefpaf) [reply recipes/pycgtool/meta.yaml]: Even when pip installing this is a workaround. Ideally the dependency should declare cython as its build dependency, or better yet, build and publish wheels. It is not a big deal for your to supply cython to make things work but it is not encouraged as a practice. I know... Python packaging is messy...

**Pull request merged**